sys/include/net/ng_ipv6/netif.h

```diff
@@ -0,0 +1,248 @@
+} ng_ipv6_netif_t;
+
+/**
+ * @brief Initializes the module.
+ */
+void ng_ipv6_netif_init(void);
+
+/**
+ * @brief   Add interface to IPv6.
+ *
+ * @details This function will be called by @ref ng_netif_add().
+ *
+ * @param[in] pid   The PID to the interface
+ */
+void ng_ipv6_netif_add(kernel_pid_t pid);
+
+/**
+ * @brief   Remove interface from IPv6.
+ *
+ * @details This function will be called by @ref ng_netif_remove().
+ *
+ * @param[in] pid   The PID to the interface
+ */
+void ng_ipv6_netif_remove(kernel_pid_t pid);
+
+/**
+ * @brief   Get interface.
+ *
+ * @param[in] pid   The PID to the interface
+ *
+ * @return  The interface describing structure, on success.
+ * @return  NULL, if there is no interface with PID @p pid.
+ */
+ng_ipv6_netif_t *ng_ipv6_netif_get(kernel_pid_t pid);
+
+/**
+ * @brief   Adds an address to an interface.
+ *
+ * @param[in] pid       The PID to the interface
+ * @param[in] addr      An address you want to add to the interface
+ * @param[in] anycast   If @p addr should be an anycast address, @p anycast
+ *                      must be true. Otherwise set it false.
+ *                      If @p addr is a multicast address, @p anycast will be
+ *                      ignored.
+ *
+ * @see <a href="https://tools.ietf.org/html/rfc4291#section-2.6">
+ *          RFC 4291, section 2.6
+ *      </a>
+ *
+ * @return  0, on success.
+ * @return  -EINVAL, if @p addr is NULL or unspecified address.
+ * @return  -ENOENT, if @p pid is no interface.
+ * @return  -ENOMEM, if there is no space left to store @p addr.
+ */
+int ng_ipv6_netif_add_addr(kernel_pid_t pid, const ng_ipv6_addr_t *addr,
+                           bool anycast);
+
+/**
+ * @brief   Remove an address from the interface.
+ *
+ * @param[in] pid       The PID to the interface
+ * @param[in] addr      An address you want to remove from interface
+ */
+void ng_ipv6_netif_remove_addr(kernel_pid_t pid, ng_ipv6_addr_t *addr);
+
+/**
+ * @brief   Removes all addresses from the interface.
+ *
+ * @param[in] pid       The PID to the interface
+ */
+void ng_ipv6_netif_reset_addr(kernel_pid_t pid);
+
+/**
+ * @brief   Searches for an address on all interfaces
+ *
+ * @param[out] out  The reference to the address on the interface
+ * @param[in] addr  The address you want to search for
+ *
+ * @return  The PID to the interface the address is registered to
+ * @return  KERNEL_PID_UNDEF, if the address can not be found on any interface
+ */
+kernel_pid_t ng_ipv6_netif_find_addr(ng_ipv6_addr_t **out, const ng_ipv6_addr_t *addr);
+
+/**
+ * @brief   Searches for an address on an interfaces
+ *
+ * @param[in] pid   The PID to the interface
+ * @param[in] addr  The address you want to search for
+ *
+ * @return  The reference to the address on the interface.
+ * @return  NULL, if the address can not be found on the interface
+ * @return  NULL, if @p pid is no interface.
+ */
+ng_ipv6_addr_t *ng_ipv6_netif_find_addr_local(kernel_pid_t pid, const ng_ipv6_addr_t *addr);
```

Comment:
I find the name `local` somewhat confusing in this context (though I know its used with the same meaning at other places). Maybe `ng_ipv6_netif_find_if()` or `ng_ipv6_netif_find_interface()` for the function above and `ng_ipv6_netif_find_addr()` for this one?